packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -109,6 +219,210 @@ abstract contract PermissionManager is Initializable {
         _initializePermissionManager({_initialOwner: _initialOwner});
     }
 
+    /// @notice Modifier used to protect PM methods from only being called by allowed owners.
+    /// @param _where The target contract to revoke or give permissions on.
+    /// @param _permissionId The permission to check the permissions for.
+    /// @param _operation The operation used to check ownership. The use of the PermissionLib.Operation type here instead of PM.Option simplifies the code (see applyTarget methods).
+    modifier ownerAuth(
+        address _where,
+        bytes32 _permissionId,
+        PermissionLib.Operation _operation
+    ) {
+        Permission storage permission = permissions[permissionHash(_where, _permissionId)];
+
+        if (_isPermissionFrozen(permission)) {
+            revert PermissionFrozen(_where, _permissionId);
+        }
+
+        if (!_checkOwner(permission, msg.sender, _operation, _isRoot(msg.sender))) {
+            revert Unauthorized(_where, msg.sender, _permissionId);
+        }
```

Comment:
Why revert `Unauthorized` instead of `UnauthorizedOwner`? 

It says `UnauthorizedOwner` should be 
> /// @notice Thrown if the calling account doesn't have the correct permission flags set.